trunner/runners/STM32L4Runner.py

```diff
@@ -0,0 +1,77 @@
+#
+# Phoenix-RTOS test runner
+#
+# Stm32l4 runner
+#
+# Copyright 2021 Phoenix SYstems
+# Authors: Mateusz Niewiadomski
+#
+
+import serial
+import time
+import pexpect.fdpexpect
+
+import sys
+from os import path
+from subprocess import Popen, PIPE
+
+from .common import DeviceRunner
+from .common import _BOOT_DIR
+
+
+class STM32L4Runner(DeviceRunner):
+
+    # oneByOne_fdspawn class is a workaround for Phoenix-RTOS on stm32l4 targets not processing characters fast enough
+    # This class inherits and is passed to harness instead of pexpect.fdpexpect.fdspawn
+    # It redefines send() with addition of timeout after each letter sent. Timeout set to 30ms per character
+    class oneByOne_fdspawn(pexpect.fdpexpect.fdspawn):
+
+        def send(self, s):
+            r = 0
+            for c in s:
+                r += super().send(c)
+                time.sleep(0.03)
+            return r
+
+    # STM32L4Runner class variables:
+    # - BINARY_NAME: name of Phoenix-RTOS image binary that should be available in _boot/ directory
+    # - OPENOCD_CMD: string with shell command calling openocd with correct parameters to flash target
+    OPENOCD_CMD = ['openocd -f interface/stlink.cfg -f target/stm32l4x.cfg -c \
+"reset_config srst_only srst_nogate connect_assert_srst" -c "program ', ' 0x08000000 verify reset exit"']
```

Comment:
The problem is that this command should include a path to binary here:
```
(...) -c "program <PATH_TO_BINARY> 0x08000000 verify (...)
```
and with that it breaks no matter what I do:
 - insert path variable as list argument
 - insert path variable by `'(...)'.format()` method
 - insert path directly into the string as plain text

I think the issue is that openocd interprets commandline arguments in some strange way. List-ifying the command breaks the `openocd` interpreting it. I have tried also making arguments like:
```
'-f interface/stlink.cfg', '-f'
```
instead of 
```
'-f', 'interface/stlink.cfg', '-f'
```
but it just produces different errors (like `openocd` cant figure out where it is launched from as it cannot find config files).

Reply:
stil, it seems You're doing something wrong, as the approach suggested by @jsarzy is the correct one 9albeit there are some errors - the parameters should not be in double `"` as this is parameter encasing for shell only) . Please share the exact code You're trying out so we could spot potential errors.


I've quickly verified that code like this works:
```python
CMD = ['openocd', '-f', 'interface/stlink.cfg', '-f', 'target/stm32l4x.cfg', '-c', "reset_config srst_only srst_nogate connect_assert_srst",  '-c',  "program {progname} 0x08000000 verify reset exit"]
cmd_run = [x.format(progname='real_program_path') for x in CMD]
Popen(cmd_run)
```

Reply:
Yes it was a problem. Removing `"` from arguments made all work as it should without `shell=True` parameter. Thanks!